pkg/apiserver/domain/service/workflow.go

```diff
@@ -368,6 +369,10 @@ func (w *workflowServiceImpl) SyncWorkflowRecord(ctx context.Context) error {
 			continue
 		}
 
+		if app.Status.Workflow == nil {
+			continue
```

Comment:
why logic changed?

Reply:
`app.Status.Workflow` is a pointer, we did not check the pointer before we refer it before.